types/react-router/index.d.ts

```diff
@@ -62,34 +63,36 @@ export interface RouteComponentProps<P> {
   staticContext?: any;
 }
 
-export interface RouteProps {
+export type RouteComponentPropsUnion<P> = Partial<RouteComponentProps<any>> & P;
+
+export interface RouteProps<P> {
   location?: H.Location;
-  component?: React.SFC<RouteComponentProps<any> | undefined> | React.ComponentClass<RouteComponentProps<any> | undefined>;
+  component?: React.SFC<RouteComponentPropsUnion<P> | undefined> | React.ComponentClass<RouteComponentPropsUnion<P> | undefined>;
```

Comment:
The `P` was added specifically to be used there. The `Partial` was intended, but perhaps not the correct approach. 

The `RouteComponentProps<any>` properties are always sent, but most third party components passed to the component property are unaware of them and rely solely on the properties in `P`.